src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1623,9 +1632,21 @@ Node* VectorInsertNode::make(Node* vec, Node* new_val, int position) {
 }
 
 Node* VectorUnboxNode::Ideal(PhaseGVN* phase, bool can_reshape) {
-  Node* n = obj()->uncast();
+  Node* n = obj();
+  // Vector APIs are lazily intrinsified, during parsing compiler emits a
+  // call to intrinsic function, since most of the APIs return an abstract vector
+  // hence a subsequent checkcast results into a graph shape comprising of CheckPP
+  // and CheckCastPP chain. During lazy inline expansion, call gets replaced by
+  // a VectorBox but we still need to traverse back through chain of cast nodes
+  // to get to the VectorBox.
+  if (n->is_InlineType() &&
```

Comment:
Since compiler will generate an InlineTypeNode for any value/primitive instance, is there any other cases that the input of `VectorUnbox` is not an `InlineTypeNode` ?  If not, maybe we could remove this condition and add an assertion instead?

Reply:
Correct, added assertion.